src/driver/amdxdna/aie2_event_trace.c

```diff
@@ -393,13 +403,67 @@ void aie2_config_event_trace(struct amdxdna_dev_hdl *ndev, u32 enable,
 	if (ret)
 		return;
 
-	XDNA_DBG(ndev->xdna, "buf size 0x%08x category 0x%08x",
-		 buff_size, category);
 	ndev->event_trace_req->dram_buffer_size = buff_size;
 	ndev->event_trace_req->event_trace_category = category;
 	aie2_assign_event_trace_state(ndev, true);
 }
 
+static void aie2_event_trace_handle_autosuspend(struct amdxdna_dev_hdl *ndev,
+						bool suspend)
```

Comment:
From the readability point of view, can we avoid using "single function + status" here?

What is expected is that,
aie2_event_trace_suspend()
{
  // acquire lock
  // sanity check
  // if event trace is enable, stop event trace
  // release lock
}

aie2_event_trace_resume()
{
  // acquire lock
  // sanity check
  // if event trace is enable, start event trace
  // release lock
}

In my test, the current code of this function is correct. But coupling suspend and resume logic in one function make it harder to read and error-prone for long term maintenance.